azerothcore/azerothcore-wotlk
# feat(Core): port aggro distance from vMaNGOS (read detection_range values from creature_template)

## Changes Proposed:
-  read detection_range values from creature_template
-  contains DB records for unique creatures (for WoW classic only)

## Issues Addressed:

- Closes #5455

## SOURCE:

https://github.com/vmangos/core/commit/9f50e90ac9ed7db233a7343b324c559e021bb5b3
https://github.com/vmangos/core/commit/af745365e9bd7f0963734510452c0ac957fb1424

## Tests Performed:

- can be tested in the game
- 


## How to Test the Changes:


1. Add SQL update
2. "The aggro radius of a mob having the same level as the player is the new column detection_range from the creature_template table"
2. Check if it fits creature range from DB (The formula for the calculation is:  same level as the player is detection_range column )
for each level up minus one yard
for each level down plus one yard

## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
